### PR TITLE
[mlir][SVE] Add more e2e test for vector.contract 

### PR DESCRIPTION
**Pull request by banach-space**
Adds basic integration tests for `vector.contract` for the dot product
and matvec operations. These tests exercise scalable vectors.

Depends on https://github.com/llvm/llvm-project/pull/69845

### CODE REVIEW

**Review by MacDue — APPROVED** [review]
LG! :)
Typo in PR description:
s/matviec/matvec/g

File: mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir [hunk 1]
```diff
@@ -30,6 +56,60 @@
...
+  %vector_a = arith.constant dense<123> : vector<[4]xi32>
+  %vector_b = arith.constant dense<314> : vector<[4]xi32>
+  %vector_c = arith.constant dense<0> : vector<[4]xi32>
+
+  // The result of this dot-product will depend
+  // on the vector length, so we are unable to verify it.
+  %dp1 = vector.contract #dotp_trait %vector_a, %vector_b, %acc
+    : vector<[4]xi32>, vector<[4]xi32> into i32
+  // DP: {{[0-9]*}}
+  vector.print %dp1 : i32
```
[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir:72]: Neat, I love this!

[the remark this answers is not in the document]
Comment (banach-space) [reply mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir]: You actually need to multiply by 4 as well (the number of elements is 4 x `vscale`). Just as an FYI :)

File: mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir [hunk 2]
```diff
@@ -30,6 +56,65 @@
...
+// Contraction: matrix-vector A x c
+func.func @matvec_i32() {
+  %acc = arith.constant dense<0>: vector<3xi32>
+
+  %vector_a = arith.constant dense<123> : vector<3x[4]xi32>
+  %vector_b = arith.constant dense<314> : vector<[4]xi32>
+  %vector_c = arith.constant dense<0> : vector<[4]xi32>
+
+  // The result of this matvec will depend on the vector length, so we are
+  // unable to verify it.
```
Comment (MacDue) [inline mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir:100]: P.s. You can do the same trick here :)

File: mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir [hunk 3]
```diff
@@ -30,6 +56,65 @@
...
+// Contraction: dot-product a x b.
+func.func @dot_product_i32() {
+  %acc = arith.constant 0: i32
+
+  %vector_a = arith.constant dense<123> : vector<[4]xi32>
+  %vector_b = arith.constant dense<314> : vector<[4]xi32>
+  %vector_c = arith.constant dense<0> : vector<[4]xi32>
+
+  // The result of this dot-product will depend
+  // on the vector length, so we are unable to verify it.
```
Comment (MacDue) [inline mlir/test/Integration/Dialect/Vector/CPU/ArmSVE/test-contraction.mlir:68]: Also this comment is now outdated :)

**Pull request merged**